test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -5230,6 +5233,20 @@ BOOST_AUTO_TEST_CASE(storage_string_as_mapping_key_without_variable)
 	BOOST_CHECK(callContractFunction("f()") == encodeArgs(u256(2)));
 }
 
+BOOST_AUTO_TEST_CASE(array_out_of_bound_access)
```

Comment:
Please move this to SolidityNameAndTypeResolution, as it tests the analysis phase.